bin/ch/WScriptJsrt.cpp

```diff
@@ -225,33 +238,39 @@ JsValueRef WScriptJsrt::LoadScriptHelper(JsValueRef callee, bool isConstructCall
     return returnValue;
 }
 
-JsValueRef WScriptJsrt::LoadScript(JsValueRef callee, LPCWSTR fileName, size_t fileNameLength, LPCWSTR fileContent, LPCWSTR scriptInjectType, bool isSourceModule)
+JsValueRef WScriptJsrt::LoadScript(JsValueRef callee, LPCSTR fileName, LPCWSTR fileContent, LPCWSTR scriptInjectType, bool isSourceModule)
 {
     HRESULT hr = E_FAIL;
     JsErrorCode errorCode = JsNoError;
-    LPCWSTR errorMessage = L"Internal error.";
+    LPCWSTR errorMessage = _u("Internal error.");
     size_t errorMessageLength = wcslen(errorMessage);
     JsValueRef returnValue = JS_INVALID_REFERENCE;
     JsErrorCode innerErrorCode = JsNoError;
     JsContextRef currentContext = JS_INVALID_REFERENCE;
     JsRuntimeHandle runtime = JS_INVALID_RUNTIME_HANDLE;
-
+    
+    wchar_t* fullPath = nullptr;
+    char fullPathNarrow[_MAX_PATH];
+    size_t len = 0;
+    
     IfJsrtErrorSetGo(ChakraRTInterface::JsGetCurrentContext(&currentContext));
     IfJsrtErrorSetGo(ChakraRTInterface::JsGetRuntime(currentContext, &runtime));
 
-    wchar_t fullPath[_MAX_PATH];
-    if (_wfullpath(fullPath, fileName, _MAX_PATH) == nullptr)
+    if (_fullpath(fullPathNarrow, fileName, _MAX_PATH) == nullptr)
     {
         IfFailGo(E_FAIL);
     }
     // canonicalize that path name to lower case for the profile storage
-    size_t len = wcslen(fullPath);
+    // REVIEW: This doesn't work for UTF8...
+    len = strlen(fullPathNarrow);
```

Comment:
I know windows has `_mbstrlen` but a quick look around doesn't reveal an obvious Linux version of the same, but there are apparently libraries like UTF8-CPP that would contain the requisite function. I'm not sure whether it would be necessary to know *a priori* whether the string is ASCII or UTF-8 since ASCII characters with binary sequences starting with 1{1,4}0 could be interpreted as part of a UTF-8 multi-byte sequence.